packages/axios-extension/src/abap/abap-service-provider.ts

```diff
@@ -288,6 +290,38 @@ export class AbapServiceProvider extends ServiceProvider {
         return generator;
     }
 
+    /**
+     * Collects ValueListReferences annotation values from the service metadata and annotation files.
+     *
+     * @param references - Service references for which metadata should be fetched.
+     * @returns A list of ValueListReferences found in the metadata and annotations.
+     */
+    public async fetchValueListReferenceServices(
+        references: { target: string; rootPath: string; value: string }[]
+    ): Promise<{ target: string; data?: string; path: string }[]> {
```

Comment:
Can we please use the same type as input here as is returned by `getValueListReferences()`. I think the type can be defined in axios extension and add a dev dep. to odata-service-writer

Reply:
Good catch, I removed all the consumers so typescript didn't pick up this mismatch, but this should not be an issue when we have consuming code.

I didn't want to add the dependency to the `odata-service-writer`, because usually it is a slippery slope and it would make it easier to add other unintended usages from the `axios-extension`.

If we use duck types, the main downside is that it is more difficult to see the link between `getValueListReferences` output type and `fetchValueListReferenceServices` input type. However, changing either of them still would make typescript compiler to complain at the consumption point.

I think this is the best of the not so great options, but if you insist I'll move the type to the `axios-extension`.